### ISSUE TITLE
Cannot run TridentNet onto GPU device - no kernel image is available for execution on the device

### ISSUE THREAD

**Issue by vedrusss** [TuSimple/simpledet#115]
Hi there,
I spend almost a week looking for how to run the detector onto GPU.
I do it onto amazon cloud instance with following host env:
Ubuntu **16.04**
Python **3.5**
Cuda **10.1**
GPU driver **418.40.04**
GPU device **Tesla K80** with 11441Mb memory.

To run the TridentNet I **used both** approaches:
**1)** I installed SimpleDet framework onto the host with building MXNet from scratch **as described in install.md** (with copying custom operators); I did `git clone https://github.com/RogerChern/mxnext` and run `make` command within simpledet folder;
**2)** I used provided docker image **rogerchen/simpledet:cuda10**; I did `git clone https://github.com/RogerChern/mxnext` and run `make` command within simpledet folder;

During both approaches I tried to run modified detection_test.py script (modification was made to exclude coco but run the detector against a single image). _The script attached_.
[detection_test_modified.txt](https://github.com/TuSimple/simpledet/files/3258027/detection_test_modified.txt)

All **worked** fine in **CPU** mode, but slowly.

In **GPU** mode I'm facing next **error** which becomes my dream in night because I can't get around it (the error example was obtained within _docker container_):
```
root@a87357781b9e /simpledet# python detection_test_to_savant_json.py --config config/tridentnet_r101v2c4_c5_addminival_2x.py --test demo/1124AM_person_not_car.209.png                         134 master?
load experiments/tridentnet_r101v2c4_c5_addminival_2x/checkpoint-0012.params
(352, 640, 3)
800 1455 2.272727272727273
len of outputs 5
Traceback (most recent call last):
  File "detection_test_to_savant_json.py", line 128, in <module>
    outputs = [x.asnumpy() for x in results]
  File "detection_test_to_savant_json.py", line 128, in <listcomp>
    outputs = [x.asnumpy() for x in results]
  File "/root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/mxnet-1.5.0-py3.6.egg/mxnet/ndarray/ndarray.py", line 1995, in asnumpy
    ctypes.c_size_t(data.size)))
  File "/root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/mxnet-1.5.0-py3.6.egg/mxnet/base.py", line 254, in check_call
    raise MXNetError(py_str(_LIB.MXGetLastError()))
mxnet.base.MXNetError: [15:53:39] /tmp/mxnet/3rdparty/mshadow/mshadow/././././cuda/tensor_gpu-inl.cuh:110: Check failed: err == cudaSuccess (48 vs. 0) : Name: MapPlanKernel ErrStr:no kernel image is available for execution on the device
Stack trace:
  [bt] (0) /root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/mxnet-1.5.0-py3.6.egg/mxnet/libmxnet.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x32) [0x7f67c0319da2]
  [bt] (1) /root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/mxnet-1.5.0-py3.6.egg/mxnet/libmxnet.so(void mshadow::cuda::MapPlan<mshadow::sv::saveto, mshadow::Tensor<mshadow::gpu, 2, float>, mshadow::expr::ScalarExp<float>, float>(mshadow::expr::Plan<mshadow::Tensor<mshadow::gpu, 2, float>, float>, mshadow::expr::Plan<mshadow::expr::ScalarExp<float>, float> const&, mshadow::Shape<2>, CUstream_st*)+0x1ec) [0x7f67c398345c]
  [bt] (2) /root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/mxnet-1.5.0-py3.6.egg/mxnet/libmxnet.so(void mxnet::ndarray::Eval<mshadow::gpu>(float const&, mxnet::TBlob*, mxnet::RunContext)+0x167) [0x7f67c3bb9377]
  [bt] (3) /root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/mxnet-1.5.0-py3.6.egg/mxnet/libmxnet.so(+0x288dc61) [0x7f67c1c2cc61]
  [bt] (4) /root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/mxnet-1.5.0-py3.6.egg/mxnet/libmxnet.so(+0x288dd07) [0x7f67c1c2cd07]
  [bt] (5) /root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/mxnet-1.5.0-py3.6.egg/mxnet/libmxnet.so(+0x2d3cf8e) [0x7f67c20dbf8e]
  [bt] (6) /root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/mxnet-1.5.0-py3.6.egg/mxnet/libmxnet.so(mxnet::engine::ThreadedEngine::ExecuteOprBlock(mxnet::RunContext, mxnet::engine::OprBlock*)+0x945) [0x7f67c20d6275]
  [bt] (7) /root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/mxnet-1.5.0-py3.6.egg/mxnet/libmxnet.so(void mxnet::engine::ThreadedEnginePerDevice::GPUWorker<(dmlc::ConcurrentQueueType)0>(mxnet::Context, bool, mxnet::engine::ThreadedEnginePerDevice::ThreadWorkerBlock<(dmlc::ConcurrentQueueType)0>*, std::shared_ptr<dmlc::ManualEvent> const&)+0x11d) [0x7f67c20edc2d]
  [bt] (8) /root/.pyenv/versions/3.6.8/lib/python3.6/site-packages/mxnet-1.5.0-py3.6.egg/mxnet/libmxnet.so(std::_Function_handler<void (std::shared_ptr<dmlc::ManualEvent>), mxnet::engine::ThreadedEnginePerDevice::PushToExecute(mxnet::engine::OprBlock*, bool)::{lambda()#4}::operator()() const::{lambda(std::shared_ptr<dmlc::ManualEvent>)#1}>::_M_invoke(std::_Any_data const&, std::shared_ptr<dmlc::ManualEvent>&&)+0x4e) [0x7f67c20edede]
```

What I noticed using 'nvidia-smi' during test run: 
python process allocates ~406 Mb of GPU at the beginning. But then test script just crashes with the error provided above.

Please, help! It really makes me crazy

**Comment by RogerChern** [comment]
Really sorry about this. We only generate binary codes for Maxwell and later GPU archs when building current docker images. A quick fix would be rebuilt mxnet with compute_35 arch as below. We will update the docker image soon to reflect this change. Sorry for the inconvenience.

```
git clone --recursive https://github.com/apache/incubator-mxnet /tmp/mxnet && \
git clone https://github.com/Tusimple/simpledet /tmp/simpledet && \
git clone https://github.com/RogerChern/cocoapi /tmp/cocoapi && \
cp -r /tmp/simpledet/operator_cxx/* /tmp/mxnet/src/operator && \
mkdir -p /tmp/mxnet/src/coco_api && \
cp -r /tmp/cocoapi/common /tmp/mxnet/src/coco_api && \
cd /tmp/mxnet && \
echo "USE_OPENCV = 0" >> ./config.mk && \
echo "USE_MKLDNN = 0" >> ./config.mk && \
echo "USE_BLAS = openblas" >> ./config.mk && \
echo "USE_CUDA = 1" >> ./config.mk && \
echo "USE_CUDA_PATH = /usr/local/cuda" >> ./config.mk && \
echo "USE_CUDNN = 1" >> ./config.mk && \
echo "USE_NCCL = 1" >> ./config.mk && \
echo "USE_DIST_KVSTORE = 1" >> ./config.mk && \
echo "CUDA_ARCH = -gencode arch=compute_35,code=sm_35" >> ./config.mk && \
make -j$((`nproc`-1)) && \
cd python && \
python3 setup.py install && \
rm -rf /tmp/mxnet /tmp/simpledet /tmp/cocoapi
```

**Comment by vedrusss** [comment]
Thanks, @RogerChern , that worked for me. Finally! I'm so happy. 
But I must notice you to make it work I had to use older commit of mxnet
`https://github.com/RogerChern/mxnext.git 71fd7f551c70f8584acc706c3ea9945f3f23b078`
because the latest one didn't work for me neither in gpu nor in cpu mode - I had obtained error like
`operands could not be broadcast together with shapes version [1,1000,3] and [1,3]`
I can't provide now the exact error output because I've already switched to older mxnext version (excluding commits of last week).
So, pleases take a look at that.

Thanks much for your advice. I had such idea, to add lower arch to build, but tried once and wrong version sm_40.
